buckets-mdapi/src/object/get.rs

```diff
@@ -64,33 +76,40 @@ pub(crate) fn action(
 fn do_get(
     method: &str,
     payload: &GetObjectPayload,
-    mut conn: &mut PostgresConnection,
+    conn: &mut PostgresConnection,
     metrics: &RegisteredMetrics,
     log: &Logger,
-) -> Result<Option<ObjectResponse>, String> {
-    let sql = get_sql(payload.vnode);
+) -> Result<Option<ObjectResponse>, Value> {
+    let mut txn = (*conn).transaction().map_err(|e| e.to_string())?;
+    let get_sql = sql::get_sql(payload.vnode);
 
-    sql::query(
-        sql::Method::ObjectGet,
-        &mut conn,
-        sql.as_str(),
+    conditional::request(
+        &mut txn,
         &[&payload.owner, &payload.bucket_id, &payload.name],
+        payload.vnode,
+        &payload.conditions,
         metrics,
         log,
     )
-    .map_err(|e| e.to_string())
-    .and_then(|rows| response(method, &rows))
-}
+    .and_then(|_rows| {
+        // XXX
+        //
+        // Somehow this should skip this second get if the above conditional already gets the
```

Comment:
I started looking at this, but I wasn't totally keen on having to also check for 0 or >1 rows in `do_get`, similar to what `conditional::request` has to do.

I've made some changes in https://github.com/joyent/manta-buckets-mdapi/pull/47/commits/fe7e575d15bc9def62a033529ec322712020153e that causes `conditional::request` to return the rows it finds in its query, and then `go_get` will just pass those back if there are `Some(rows)`. Otherwise, `None` causes `go_get` to do the query itself (which would be the case if the request wasn't conditional).

I _think_ it's doing the right thing, but I'd appreciate eyes on it!